src/cargo/sources/git/source.rs

```diff
@@ -169,7 +169,10 @@ impl<'cfg> Source for GitSource<'cfg> {
                 format!("git repository `{}`", self.remote.url())));
 
             trace!("updating git source `{:?}`", self.remote);
-            let repo = try!(self.remote.checkout(&self.db_path));
+
+            let repo = try!(network::with_retry(&self.config, ||{
+              self.remote.checkout(&self.db_path)
```

Comment:
Im going to need some advice here. I started to put a  `cargo_config: &'cfg Config` attribute on `GitRemote`. This lead to a cascade of lifetime markers. After that cleared up I am getting more errors. I think I can reason about the errors but I just wanted to make sure my approach was right. 
Thanks!
Becker


[current list of errors](https://gist.github.com/sbeckeriv/3171ff0b46588c49fd51)